src/librustc_mir/borrow_check/places_conflict.rs

```diff
@@ -17,17 +17,50 @@ use rustc::mir::{Projection, ProjectionElem};
 use rustc::ty::{self, TyCtxt};
 use std::cmp::max;
 
+/// When checking if a place conflicts with another place, this enum is used to influence decisions
+/// where a place might be equal or disjoint with another place, such as if `a[i] == a[j]`.
+/// `PlaceConflictBias::Overlap` would bias toward assuming that `i == j` and that these places
```

Comment:
small nit: Would it not be more precise to say "`Overlap` would bias towards assuming that `i` *might* equal `j`, and thus that these place could overlap. `NoOverlap` assumes that `i != j` and that this places must be disjoint."

The key change being adding the word "might" up above. I don't know if it would ever be relevant, but its good when reading the code to understand that e.g. you should not provide diagnostics that assume nor assert that `i` and `j` *are* equal.

Reply:
On second thought, I'm not sure I'm right in claiming that `NoOverlap` is stating that the compared indices *must* be disjoint; rather, it is saying something like: "for the purposes of the predicate being run in the calling context, the conservative choice is to assume the compared indices are disjoint."

Right...? Anyway I'll have more comment on this on a use of `NoOverlap` below.